### PR TITLE
Untested files

### PR DESCRIPTION
**Pull request by EWhite613**
Require untested files.

### CODE REVIEW

File: lib/templates/test-body-footer.html [hunk 1]
```diff
@@ -36,8 +36,42 @@
...
   if (typeof Testem !== "undefined" && Testem.afterTests) {
     Testem.afterTests(function(config, data, callback) {
+      var seen = {}
+      var appName = getModuleName().appName
+      for (var moduleName in requirejs.entries) {
+        if (typeof(seen[moduleName]) === 'undefined') {
+          var parts = moduleName.split('/')
+            try {
+              if (parts.length >= 1 && parts[0] === appName || parts.length >= 2 && parts[0] === 'dummy' && parts[1] === appName){
+                  if (!shouldExclude(moduleName).result){
```
Comment (rwjblue) [inline lib/templates/test-body-footer.html]: Can we flip this on its head: `POST` the result of `Object.keys(requirejs.entries)` and have the server return an array of modules to require?

**Comment by mike183** [comment]
Just given your branch a try. From a glance, it appears to be working very well!

 👍

**Comment by rwjblue** [comment]
Where are we with this?  

@sglanzer - Can you take a peek?

**Comment by sandersky** [comment]
@EWhite613 I'd love to see more tests around all of the code this pull request adds.

**Comment by sglanzer** [comment]
@rwjblue @sandersky 

Sorry, these comments got buried in my email, I'll take some time today to review

File: lib/attach-middleware.js [hunk 2]
```diff
@@ -34,4 +34,41 @@ module.exports = function(app, options) {
...
+  }
+  app.post('/filter-entries',bodyParser.urlencoded({
+    limit: '50mb',
+    extended: false,
+    parameterLimit: 10000
+  }),bodyParser.json({
+    limit: '50mb'
+  }),
+    function(req, res) {
+      var entries = req.body['entries[]'] || req.body.entries
```
Comment (sglanzer) [inline lib/attach-middleware.js]: Can you comment on the need for this?

File: lib/attach-middleware.js [hunk 3]
```diff
@@ -34,4 +34,41 @@ module.exports = function(app, options) {
...
+      debugger
+      var seen = {}
+      var appName = options.appName
+      var filteredEntries = entries.filter(function(entry) {
```
Comment (sglanzer) [inline lib/attach-middleware.js]: Can the filter be handled in the middleware so that we only get a list of entries that weren't loaded?

File: lib/attach-middleware.js [hunk 4]
```diff
@@ -34,4 +34,41 @@ module.exports = function(app, options) {
...
+        var result = false
+        if (typeof (seen[entry]) === 'undefined') {
+          var parts = entry.split('/')
+          if (parts.length >= 1 && parts[0] === appName || parts.length >= 2 && parts[0] === 'dummy' && parts[1] === appName) {
```
Comment (sglanzer) [inline lib/attach-middleware.js]: I'm wary of this - @rwjblue is this a predictable/consistent format that can be relied on?

File: lib/coverage-instrumenter.js [hunk 5]
```diff
@@ -97,7 +97,11 @@ CoverageInstrumenter.prototype.processString = function(content, relativePath) {
 
   relativePath = fixPath(relativePath, this._appName, this._appRoot, this._templateExtensions);
 
-  return instrumenter.instrumentSync(content, relativePath);
+  try {
+    return instrumenter.instrumentSync(content, relativePath);
+  } catch (e) {
+    console.error('Unable to cover:', relativePath, '. Try setting useBabelInstrumenter to true. \n', e.stack);
```
Comment (sglanzer) [inline lib/coverage-instrumenter.js]: Can you separate this into another PR?

File: lib/templates/test-body-footer.html [hunk 6]
```diff
@@ -36,8 +36,35 @@
     });
   }
 
+  function getModuleName (){
+    return jQuery.ajax({
```
Comment (sglanzer) [inline lib/templates/test-body-footer.html]: I'd prefer not to rely directly on jQuery, if ember-cli includes an ajax lib let's use that - feel free to tell me that it's jQuery ;)

File: lib/templates/test-body-footer.html [hunk 7]
```diff
@@ -36,8 +36,35 @@
...
+      url: '/filter-entries',
+      async: false,
+      data: {
+       entries: entries 
+      }
+    }).responseJSON
+  }
   if (typeof Testem !== "undefined" && Testem.afterTests) {
     Testem.afterTests(function(config, data, callback) {
+      console.log('filtering entries')
```
Comment (sglanzer) [inline lib/templates/test-body-footer.html]: Either remove or use the `ember-cli` logger

File: test/unit/index-test.js [hunk 8]
```diff
@@ -93,16 +98,19 @@ describe('index.js', function() {
...
         expect(app.post.callCount).to.equal(0);
       });
+      it('does not GET endpoint to app', function() {
+        expect(app.get.callCount).to.equal(0);
+      });
     });
   });
 
-  describe('_doesFileExistInCurrentProject', function() {
+  describe('_doesFileExistInCurrentProjectApp', function() {
```
Comment (sglanzer) [inline test/unit/index-test.js]: Is the rename relevant to this PR?

**Comment by sglanzer** [comment]
Addresses #20

File: test/unit/index-test.js (hunk 8 above)
Comment (EWhite613) [inline test/unit/index-test.js]: No. I think I might have snuck it in the previous as well. Since `doesFileExistInCurrentProject1` was renamed to `doesFileExistInCurrentProject` -> `doesFileExistInCurrentProjectApp` but the tests weren't

**Comment by rwjblue** [comment]
https://github.com/kategengler/ember-cli-code-coverage/pull/62 landed basic support for #20.  I believe that this PR may still be needed for more thorough support, but can y'all take a look?

**Comment by sglanzer** [comment]
@rwjblue looking now

**Comment by sglanzer** [comment]
@rwjblue this PR also includes filtering code (has this been addressed?), but the code related to covering un-tested files looks to be addressed in #62 

I'm assuming https://github.com/kategengler/ember-cli-code-coverage/pull/62/files#diff-94eab975fb48e219fc130d82e21457acR6 will work the same for addon / app paths

Anyhow, I'm good to close this and port code into a filter PR (if required)

**Comment by rwwagner90** [comment]
@rwjblue @sglanzer I can help with porting anything that may still be desired from this PR, but AFAIK, we now have support for pulling in un-tested files.

**Comment by sglanzer** [comment]
Close it up

**Comment by rwwagner90** [comment]
Going to close this, feel free to reopen if it is deemed we still want to try to merge. Otherwise, please advise of the parts we want to pull out of this, and I can help get them in.
